tests/phpunit/tests/blocks/renderBlock.php

```diff
@@ -192,4 +192,136 @@ public function test_default_context_is_filterable() {
 		$this->assertSame( array( 'example' => 'ok' ), $provided_context[0] );
 	}
+
+	/**
+	 * Tests the behavior of the 'render_block_context' filter based on the location of the filtered block.
+	 *
+	 * @ticket 62046
+	 */
+	public function test_render_block_context_inner_blocks() {
+		$provided_context = array();
+
+		register_block_type(
+			'tests/context-provider',
+			array(
+				'provides_context' => array( 'example' ),
+			)
+		);
+
+		register_block_type(
+			'tests/context-consumer',
+			array(
+				'uses_context'    => array( 'example' ),
+				'render_callback' => static function ( $attributes, $content, $block ) use ( &$provided_context ) {
+					$provided_context = $block->context;
+
+					return '';
+				},
+			)
+		);
+
+		// Filter the context provided by the test block.
+		add_filter(
+			'render_block_context',
+			function ( $context, $parsed_block ) {
+				if ( isset( $parsed_block['blockName'] ) && 'tests/context-provider' === $parsed_block['blockName'] ) {
+					$context['example'] = 'ok';
+				}
+
+				return $context;
+			},
+			10,
+			2
+		);
+
+		// Test inner block context when the provider block is a top-level block.
+		do_blocks(
+			<<<HTML
+<!-- wp:tests/context-provider -->
+<!-- wp:tests/context-consumer /-->
+<!-- /wp:tests/context-provider -->
+HTML
+		);
+		$this->assertTrue( isset( $provided_context['example'] ), 'Test block is top-level block: Context should include "example"' );
+		$this->assertSame( 'ok', $provided_context['example'], 'Test block is top-level block: "example" in context should be "ok"' );
+
+		// Test inner block context when the provider block is an inner block.
+		do_blocks(
+			<<<HTML
+<!-- wp:group {"layout":{"type":"constrained"}} -->
+<!-- wp:tests/context-provider -->
+<!-- wp:tests/context-consumer /-->
+<!-- /wp:tests/context-provider -->
+<!-- /wp:group -->
+HTML
+		);
+		$this->assertTrue( isset( $provided_context['example'] ), 'Test block is inner block: Block context should include "example"' );
+		$this->assertSame( 'ok', $provided_context['example'], 'Test block is inner block: "example" in context should be "ok"' );
+	}
+
+	/**
+	 * Tests that the 'render_block_context' filter arbitrary context.
+	 *
+	 * @ticket 62046
+	 */
+	public function test_render_block_context_allowed_context() {
+		$provided_context = array();
+
+		register_block_type(
+			'tests/context-consumer',
+			array(
+				'uses_context'    => array( 'example' ),
+				'render_callback' => static function ( $attributes, $content, $block ) use ( &$provided_context ) {
+					$provided_context = $block->context;
+
+					return '';
+				},
+			)
+		);
+
+		add_filter(
+			'get_block_type_uses_context',
+			function ( $uses_context, $block_type ) {
+				if ( 'tests/context-consumer' === $block_type->name ) {
+					// Use array_values to reset the array keys.
+					return array_values( array_unique( array_merge( $uses_context, array( 'arbitrary' ) ) ) );
+				}
+				return $uses_context;
+			},
+			10,
+			2
+		);
```

Comment:
@mukeshpanchal27 What's the purpose of this filter? It wasn't part of the test in #7344.

Reply:
Inspiration taken from current block binding implementation https://github.com/WordPress/wordpress-develop/blob/6.6/src/wp-includes/class-wp-block-bindings-registry.php#L193-L204

Reply:
Normally, when the context is set, it's stored in the private variable `$this->available_context`, which we can't access directly. In the `update_available_context` method, it updates the `$this->context` property if the `$this->block_type->uses_context` array contains the context value, thus allowing it.

Reply:
In https://github.com/WordPress/wordpress-develop/pull/7522/commits/218b8879362939c27179dc18de7beab3a1dceab4 i remove the filter and incorporate the `uses_context`

Reply:
If I'm reading correctly, I think that means the test no longer covers the use case that was raised in the other PR, [starting with this comment](https://github.com/WordPress/wordpress-develop/pull/7344#issuecomment-2382991949).

According to that discussion, `render_block_context` should be able to supply context properties even when those properties aren't explicitly supported by the block. 

The `test_render_block_context_allowed_context()` test in the other PR is meant to verify that case by filtering in the `arbitrary` property even when the block declares support for only the `example` property. By adding `arbitrary` to `uses_context` here, I think the nature of the test has changed.